test/LayoutSpec.js

```diff
@@ -1,156 +1,104 @@
+import assert from 'node:assert';
 import fs from 'node:fs';
-import { fileURLToPath } from 'node:url';
-import { dirname } from 'node:path';
+import path from 'node:path';
+import url from 'node:url';
 
 import { layoutProcess } from 'bpmn-auto-layout';
 
-const __filename = fileURLToPath(import.meta.url);
-const __dirname = dirname(__filename);
-const outputDirectory = __dirname + '/generated/';
+const __filename = url.fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
 
+const fixturesDirectory = path.join(__dirname, 'fixtures');
+const outputDirectory = path.join(__dirname, 'output');
+const snapshotsDirectory = path.join(__dirname, 'snapshots');
 
-describe('Layout', function() {
+const UPDATE_SNAPSHOTS = process.env.UPDATE_SNAPSHOTS === 'true';
 
-  const filenames = fs.readdirSync('test/fixtures');
+describe('Layout', function() {
 
   before(() => {
-
-    // Clean result directory
     fs.rmSync(outputDirectory, { recursive: true, force: true });
     fs.mkdirSync(outputDirectory, { recursive: true });
-  });
-
-  filenames.forEach(filename => {
-
-    const iit = getIt(filename);
-
-    iit('should layout ' + filename, async function() {
-      const xml = fs.readFileSync('test/fixtures/' + filename, 'utf8');
-
-      const result = await layoutProcess(xml);
 
-      fs.writeFileSync(outputDirectory + filename, result, 'utf8');
-    });
+    if (UPDATE_SNAPSHOTS) {
+      fs.rmSync(snapshotsDirectory, { recursive: true, force: true });
+      fs.mkdirSync(snapshotsDirectory, { recursive: true });
+    }
   });
 
+  fs.readdirSync(fixturesDirectory)
+    .filter(fileName => fileName.endsWith('.bpmn'))
+    .forEach(fileName => {
+      getIt(fileName)(`should layout ${ fileName }`, async function() {
 
-  after(() => {
-
-    const generatedFiles = fs.readdirSync(outputDirectory);
+        // given
+        const xml = fs.readFileSync(path.join(fixturesDirectory, fileName), 'utf8');
 
-    const generatedDiagrams = generatedFiles.filter(f => f.endsWith('.bpmn'));
+        // when
+        const output = await layoutProcess(xml);
 
-    const originalContents = generatedDiagrams.map(
-      diagram => fs.readFileSync(__dirname + '/fixtures/' + diagram, 'utf8')
-    );
+        if (fs.existsSync(path.join(snapshotsDirectory, fileName))) {
+          const snapshot = fs.readFileSync(path.join(snapshotsDirectory, fileName), 'utf8');
 
-    const generatedContents = generatedDiagrams.map(
-      diagram => fs.readFileSync(outputDirectory + diagram, 'utf8')
-    );
+          // then
+          assert.strictEqual(output, snapshot, `Snapshot does not match output for ${ fileName }`);
+        }
 
-    const generated = generatedContents.map((contents, idx) => {
+        fs.writeFileSync(path.join(outputDirectory, fileName), output, 'utf8');
```

Comment:
Note that the result is not written if the test fails. I'd suggest to write it before the assertion.

Reply:
Good point!